hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/io/storage/row/HoodieRowParquetWriteSupport.java

```diff
@@ -46,13 +53,32 @@
   public HoodieRowParquetWriteSupport(Configuration conf, StructType structType, BloomFilter bloomFilter, HoodieWriteConfig writeConfig) {
     super();
     Configuration hadoopConf = new Configuration(conf);
-    hadoopConf.set("spark.sql.parquet.writeLegacyFormat", writeConfig.parquetWriteLegacyFormatEnabled());
+    hadoopConf.set("spark.sql.parquet.writeLegacyFormat", findSmallPrecisionDecimalType(structType) ? "true" : writeConfig.parquetWriteLegacyFormatEnabled());
```

Comment:
If `findSmallPrecisionDecimalType` returns false and `parquetWriteLegacyFormatEnabled` is set to true in the config then `spark.sql.parquet.writeLegacyFormat` will be true. Is that the intention?

Reply:
@codope  thanks for your review。
yes，I think if findsmallprecisiondecimaltype returns false, we need respect the user's settings； if findsmallprecisiondecimaltype returns true，we need ignore user's choice, because the user's choosie may lead  the failure of subsequent updating of the Hudi table

Reply:
Can we move the determination of right value for "spark.sql.parquet.writeLegacyFormat" to some higher layer. just once per driver. and also, lets add a warn log just incase we are overriding user config (parquetWriteLegacyFormatEnabled).

Reply:
good suggestion， fixed